### PR TITLE
COV-13 Add database tests with Testcontainers

### PR DESCRIPTION
**Pull request by emiliyank**

### CODE REVIEW

File: src/test/java/com/covid/api/integration/CountryRepositoryIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+    @Test
+    void whenSaveCountry_thenReturnCountry() {
+        // GIVEN
+        List<Country> countries = getTestCountries();
+        Country country = countries.get(0);
+
+        // WHEN
+        countryRepository.save(country);
+        Optional<Country> savedCountry = countryRepository.findByCountryCode("US");
```
Comment (cupakob) [inline src/test/java/com/covid/api/integration/CountryRepositoryIntegrationTest.java:44]: just for info (no need to be changed) - the when block should contains only the call of the method what are you testing. Line 41 goes to the THEN block.

File: src/test/resources/db/changelog/1-create-table-country-tests.sql [hunk 2]
```diff
@@ -0,0 +1,13 @@
+CREATE TABLE country(
```
Comment (cupakob) [inline src/test/resources/db/changelog/1-create-table-country-tests.sql]: ok, i expected here something different:

this script should contains some insert statements for dummy data. Creating table should comes from the migration for the service and with this migration here the db should be extended a for the tests purposes only. Lets talk about if you have questions.

Comment (emiliyank) [reply src/test/resources/db/changelog/1-create-table-country-tests.sql]: That's how I tried initially. But H2 syntax for creating tables is different than MySQL syntax. So this CREATE query is a bit different than the MySQL one.

Comment (cupakob) [reply src/test/resources/db/changelog/1-create-table-country-tests.sql]: okay, then we have to stay with this solution. But please put some dummy data. And be aware - maybe your tests are broken or you have to extend the tests/add more tests.

Comment (emiliyank) [reply src/test/resources/db/changelog/1-create-table-country-tests.sql]: Actually I found a way to make the syntax the same. The difference is for data types: bigint(20) and datetime . H2 does not support them but works fine with just bigint and timestamp, which is fine for MySQL too.

Comment (emiliyank) [reply src/test/resources/db/changelog/1-create-table-country-tests.sql]: And I made it work with some dummy data inserted only in the test DB. All tests succeed now

**Pull request merged**